paper-server/patches/sources/net/minecraft/world/level/block/entity/vault/VaultBlockEntity.java.patch

```diff
@@ -1,12 +1,21 @@
 --- a/net/minecraft/world/level/block/entity/vault/VaultBlockEntity.java
 +++ b/net/minecraft/world/level/block/entity/vault/VaultBlockEntity.java
-@@ -257,7 +_,12 @@
+@@ -253,11 +_,20 @@
+                 } else if (serverData.hasRewardedPlayer(player)) {
+                     playInsertFailSound(level, serverData, pos, SoundEvents.VAULT_REJECT_REWARDED_PLAYER);
+                 } else {
+-                    List<ItemStack> list = resolveItemsToEject(level, config, pos, player, stack);
++                    // Paper start - Add loot table to event
++                    LootTable lootTable = level.getServer().reloadableRegistries().getLootTable(config.lootTable());
++                    List<ItemStack> list = resolveItemsToEject(level, config, pos, player, stack, lootTable);
```

Comment:
I am a bit confused here, what is the benefit of this diff?
Does this do anything beyond "saving" a single loot table lookup against the loot table registry?

Reply:
For the Vaults yes, it just saves a lookup, however for Trial Spawners it allows you to actually check which loot table was run to generate the items, which is currently impossible since it picks from a weighted list. I'd like to be able to overwrite the items generated only for specific loot tables that are run, and this allows that.

Reply:
How does the overload on the VaultBlockEntity enable *anything* in the trial spawner?
TrialSpawner#ejectRewards directly evaluates the LootTable, the picked loot table instance is available to the BlockDispenseLootEvent there. I am a bit confused here.
Can you elaborate what you mean by this?

Reply:
Yea sorry, I probably did not describe that correctly then, mb! And yea, I don't think the double lookup is going to hurt here, registry lookups against keys are generally fast enough to not have to jump through so many hoops here.